tests/PhpParser/NodeTraverser/RectorNodeTraverserTest.php

```diff
@@ -50,6 +51,17 @@ public function testGetVisitorsForNodeWhenNoVisitorsMatch(): void
         $this->assertSame([], $visitors);
     }
 
+    public function testGetVisitorsForNodeAfterPreparation(): void
+    {
+        $class = new Class_('test');
+
```

Comment:
You can manually set `NameResolver` here for testing purpose, so should be no need to add to `RectorNodeTraverser`.

The name is is already resolved on parsing and set scope, we used `->namespacedName` that have value in the code.